util/topgen/lib.py

```diff
@@ -855,6 +855,21 @@ def num_rom_ctrl(modules):
 
 
 def find_module(modules, type):
+    '''Returns the first module of a given type
+    For ipgen modules the base template type is used for matching
```

Comment:
This can be a follow-up PR, but I think the resulting functions are a bit odd here. Would it make more sense to add an optional `use_base_template_type` argument to `find_module`?